### ISSUE TITLE
Fix rendering issues by updating katex

### ISSUE THREAD

**Issue by michamos** [inspirehep/inspirehep#1313]
We are facing issues with katex rendering: `$a$$b$`  is not recognized properly as `$a$` followed by `$b$` and is not rendering properly. @tsgit claims this is fixed in more recent versions of katex. We should try to update it and see if it fixes the issue.

**Comment by tsgit** [comment]
well, I said I'm seeing different results on the live demo at https://katex.org/
however I realize that this is simply because it doesn't use `$$` as delimiter at all

similarly for the cli

```
$ echo '\textrm{$$\mathcal{Z}$$ a $a$$b$ b}' | npx katex 
npx: installed 2 in 1.204s
/home/thorsten/.npm/_npx/36057/lib/node_modules/katex/dist/katex.js:17167
    throw error;
    ^

ParseError: KaTeX parse error: Can't use function '\mathcal' in text mode at position 11: \textrm{$$\̲m̲a̲t̲h̲c̲a̲l̲{Z}$$ a $a$$b$ …
```
clearly the `$$` is interpreted as empty math, not as start of displaymath.


I don't see an improvement with `v0.12.0` with the autorender option. This HTML 

```html
<!DOCTYPE html>
<!-- KaTeX requires the use of the HTML5 doctype. Without it, KaTeX may not render properly -->
<html>
  <head>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css" integrity="sha384-AfEj0r4/OFrOo5t7NnNe46zW/tFg\
W6x/bCJG8FqQCEo3+Aro6EYUG4+cU+KJWu/X" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.js" integrity="sha384-g7c+Jr9ZivxKLnZTDUhnkOnsh30B4H0rpLUpJ4j\
AIKs4fnJI+sEnkvrMWph2EDg4" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer
    src="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/contrib/auto-render.min.js"
    integrity="sha384-mll67QQFJfxn0IYznZYonOWZ644AWYC+Pt2cHqMaRhXVrursRwvLnLaebdGIlYNa"
            crossorigin="anonymous" onload="renderMathInElement(document.body, {delimiters: [
  {left: '$$', right: '$$', display: true},
  {left: '$', right: '$', display: false},
  {left: '\\(', right: '\\)', display: false},
  {left: '\\[', right: '\\]', display: true}
]});"></script>
  </head>
  <body>
    <h1>A note on discrete $$R$$ symmetries in $\mathbb{Z}$$_{6}$-II orbifolds with Wilson lines</h1>

  </body>
</html>
```

does not render correctly unless a space is inserted in between
the middle `$$` as in  `...$\mathbb{Z}$ $_{6}$...`

So in fact I don't think updating will help with this problem. It will be beneficial due to other bug fixes etc. since v0.10.2

**Comment by tsgit** [comment]
I filed a bug report with KaTeX. We'll see

**Comment by michamos** [comment]
This has been solved by #1828
